controllers/gce/instances/fakes.go

```diff
@@ -145,10 +158,12 @@ func (f *FakeInstanceGroups) RemoveInstancesFromInstanceGroup(name, zone string,
 	return nil
 }
 
-// AddPortToInstanceGroup fakes adding ports to an Instance Group.
-func (f *FakeInstanceGroups) AddPortToInstanceGroup(ig *compute.InstanceGroup, port int64) (*compute.NamedPort, error) {
-	f.Ports = append(f.Ports, port)
-	return &compute.NamedPort{Name: f.namer.BeName(port), Port: port}, nil
+func (f *FakeInstanceGroups) SetNamedPortsOfInstanceGroup(igName, zone string, namedPorts []*compute.NamedPort) error {
+	f.Ports = f.Ports[:0]
+	for _, port := range namedPorts {
+		f.Ports = append(f.Ports, port.Port)
+	}
+	return nil
```

Comment:
Either add it now or add a TODO. We should assert existence of the instance group and return an error if missing.